iotcore/src/main/java/com/google/android/things/iotcore/IotCoreClient.java

```diff
@@ -466,25 +540,39 @@ public void connectionLost(Throwable cause) {
 
             @Override
             public void messageArrived(String topic, MqttMessage message) {
-                if (!mConnectionParams.getConfigurationTopic().equals(topic)
-                        || onConfigurationListener == null
-                        || onConfigurationExecutor == null) {
-                    return;
+                if (mConnectionParams.getConfigurationTopic().equals(topic) && onConfigurationListener != null && onConfigurationExecutor != null) {
+                    // Call the client's OnConfigurationListener
+
+                    final byte[] payload = message.getPayload();
+                    onConfigurationExecutor.execute(
+                            new Runnable() {
+                                @Override
+                                public void run() {
+                                    onConfigurationListener.onConfigurationReceived(payload);
+                                }
+                            });
+                } else {
+                    if (topic.startsWith(mConnectionParams.getCommandsTopicPrefix()) && onCommandListener != null && onCommandExecutor != null) {
+
+                        // Call the client's OnCommandListener
+                        final byte[] payload = message.getPayload();
+                        final String subFolder = topic.replaceAll(mConnectionParams.getCommandsTopicPrefix(), "");
```

Comment:
do you intend that subfolder starts with '/'?

Reply:
yes, but that is not a requirement. What do you think would be better? With or without the `/` ?

Reply:
Yes I agree that it then feels more like a full path, I will adjust it